server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -582,21 +595,47 @@ private boolean isExpired(Subscription subscription) {
     }
 
     @Transactional
-    private void deleteExcessEntitlements(Pool existingPool) {
+    private void deleteExcessEntitlements(List<Pool> existingPools) {
         boolean lifo = !config
             .getBoolean(ConfigProperties.REVOKE_ENTITLEMENT_IN_FIFO_ORDER);
+        if (existingPools != null && !existingPools.isEmpty()) {
+            List<Pool> overFlowingPools = new ArrayList<Pool>();
+            for (Pool pool : existingPools) {
+                if (pool.isOverflowing()) {
+                    overFlowingPools.add(pool);
+                }
+            }
 
-        if (existingPool.isOverflowing()) {
-            // if pool quantity has reduced, then start with revocation.
-            Iterator<Entitlement> iter = this.poolCurator
-                .retrieveFreeEntitlementsOfPool(existingPool, lifo).iterator();
+            if (!overFlowingPools.isEmpty()) {
+                List<Entitlement> freeEntitlements = this.poolCurator.retrieveFreeEntitlementsOfPools(
+                        overFlowingPools, lifo);
+                List<Entitlement> entitlementsToDelete = new ArrayList<Entitlement>();
+                Map<String, List<Entitlement>> poolSortedEntitlements
+                  = new HashMap<String, List<Entitlement>>();
```

Comment:
This will probably all fit on one line, but if not the equals sign needs to be up with the declaration.

```
Map<String, List<Entitlement>> poolSortedEntitlements =
    new HashMap<String, List<Entitlement>>();
```